### ISSUE TITLE
Problematic distillation training batch

### ISSUE THREAD

**Issue by ravwojdyla** [Open-Athena/caduceus-distill#38]
We have stumbled upon a problematic batch in the distillation training, it causes a spike in the training and validation loss. Example from https://wandb.ai/tabtablabs/caduceus_distill/runs/biaznd28?nw=nwusertabtablabs:

<p align="center">
<img width="441" height="295" alt="Image" src="https://github.com/user-attachments/assets/0632b87d-c6a1-4d1d-8f30-b8907eaac009" />
</p> 

In this case, it takes roughly 300 updates (600 batches [^1]) to recover the loss.

Here's some facts:
 * the problematic batch is the batch number 8590
   * it's part of a global step two updates before the peak (peak -> global step 4297 [^1] in the image above)
   * the problematic update contains batches 8590 and 8591, but 8590 is the culprit 
   * if we replace batch 8590 with a random training batch the problem is gone
 * we can exactly reproduce this spike (see recipe below)

### Batch 8590

Batch 8590 is sequence `chr6:62775840-62906912`. Here's a code to read the batch 8590:

```py
from caduceus_distill.distill import DistillationDataset

# this code requires `xarray==2025.7.1` and `zarr==3.0.8`
ds = DistillationDataset("r2://cadu-distill/hg38_131_train")

bad_batch_labels, bad_batch_logits = ds[8590]
```

Apart from the AT-rich repeat segment, there's nothing immediately "wrong" with this batch:

1. no sea of `N` nucleotide
2. elevated AT counts: {T: 41543, A: 40653, C: 24720, G: 24156}

```py
plt.imshow(bad_batch_labels.reshape(2**8, -1))
```

<img width="1293" height="661" alt="Image" src="https://github.com/user-attachments/assets/b4dff5a1-aeb8-4f54-81fc-6e0382197965" />

<details><summary>Random batch for comparison</summary>
<p>

Counts: {T: 36218, A: 32024, C: 31512, G: 31318}

<img width="1295" height="664" alt="Image" src="https://github.com/user-attachments/assets/f58978be-d9fa-4505-a2b4-c08f8aa5c72b" />

</p>
</details> 

### Spike replay/reproduction

To reproduce we need to:
 * have a checkpoint before spike (it's good to have it a couple of global steps before the spike)
 * resume training from the checkpoint
 * adjust train data loader to resume from the correct batch
   * there may be a cute way to do this in Lightning, if there is, then it's not obvious (for me)
 * seed everything https://github.com/Open-Athena/caduceus-distill/blob/2d8afeee46fd41b0852fb16fcf4972e70c37e8da/caduceus_distill/distill.py#L523 

Below is roughly the code steps required to resume training in our distillation code:

#### First point `trainer.fit` at the checkpoint

```py
trainer.fit(model, train_loader, val_loader, ckpt_path=<CHECKPOINT_PATH>)
```

Let's say that the checkpoint path is `gs://cadu-distill/student-caduceus__epoch=00__val_loss_total=0.049__step=4289.ckpt`, from the checkpoint we know the upcoming global step is 4289 - we will need this to restart the training data loader.

#### Restart the training data loader

Given the upcoming global step is 4289, we need to restart the training data loader to read the 8578 batch of data (4289 * 2 [^1]), afaiu this is something that we need to add to the `DistillationDataset`, e.g.:

```diff
@@@ -137,10 -137,10 +160,14 @@@ class DistillationDataset(Dataset[EXAMP
          self,
          zarr_path: str,
          skip_batches: list[int] | None = None,
++        start_idx: int = 0,
      ) -> None:
          self.zarr_path = zarr_path
          self.ds: xr.Dataset | None = None
          self.skip_batches = skip_batches
++        self.start_idx = start_idx
@@@ -176,10 -176,10 +203,15 @@@
          return self._len

      def __getitem__(self, idx: int) -> EXAMPLE_T:
++        idx = idx + self.start_idx
@@@ -524,7 -524,7 +562,10 @@@ def main
      L.seed_everything(42, workers=True)

      # Initialize datasets
--    train_dataset = DistillationDataset(zarr_path_train)
++    train_dataset = DistillationDataset(zarr_path_train, start_idx=4289 * 2)
```

CC: @eric-czech 

[^1]: gradient accumulation = 2
